Promo card CTA: remove invalid prop on DOM element

monsieur-z opened the pull request:
#### Changes proposed in this Pull Request

In `PromoCardCta`, a lingering `selfTarget` prop passed as an attribute to an HTML element triggered a React warning in the console (`selfTarget` not being a valid attribute).

![image](https://user-images.githubusercontent.com/1620183/84817747-785cb500-afe3-11ea-98eb-8caa0685d01d.png)

This commit makes sure it's not passed anymore.

#### Testing instructions

1. Make sure you've got a WP site with a Free plan
2. Download the PR and boot Calypso
3. Make sure to enable Jetpack sections with the query param `?flags=jetpack/features-section`
4. Select the free plan site
5. Go to the _Jetpack > Backup_ section
6. Check there's no React error in the console
7. Click on the _Upgrade to business plan_ button and check that you're still in the same tab
8. Repeat the operation with the _Jetpack > Scan_ section

Commit:
Promo card CTA: remove invalid HTML attribute

In PromoCardCta, a lingering `selfTarget` prop passed as an attribute to
an HTML elememt triggered a console warning. This commit makes sure it's
not passed anymore.

Commit:
Promo card CTA: remove invalid HTML attribute

In PromoCardCta, a lingering `selfTarget` prop passed as an attribute to
an HTML elememt triggered a console warning. This commit makes sure it's
not passed anymore.

ChaosExAnima commented on client/components/promo-section/promo-card/cta.tsx:
```diff
@@ -50,25 +50,28 @@ export interface Props {
 }
 
-function isCtaAction( action: any ): action is CtaAction {
+function isCtaAction( action: any ): boolean {
```
This argument isn't any, it is a union type of `URL | ClickCallback | CtaAction` going by the code above.

Additionally, `action is CtaAction` is a very specific return type that indicates to TS that a given action is a CtaAction. With this code, I see many of the following or similar:
```
Property 'selfTarget' does not exist on type 'string | CtaAction | ClickCallback'.
```

rcanepa committed:
Jetpack section: hide Activity Log promo from the upsell page for Personal and Premium plan owners (#43335)

* Hide the AL promo in Personal and Premium plan

* Hide AL promo in Scan when Personal or Premium

ChaosExAnima reviewed (changes requested):
This code has multiple side effects and is causing a lot of errors when I review it with TS warnings in my environment.

I think this PR should be simplified to target just the removal of `selfTarget` instead of declaring more constants and changing the inner workings. This component could do with a refactor in general honestly, but that should be a different PR imo.

klimeryk committed:
Domains: Update multi level TLD list (#43351)

Originally introduced in #18648. Looks like we did not update this list as we onboarded other second level TLDs. As a result, users buying domains via /domains got the wrong product assigned.

ockham committed:
Config: Allow setting feature flags thru a qarg in wpcalypso and horizon (#43345)

Setting feature flags by appending a `?flags=` query arg to a Calypso URL has been possible for the `development`, `stage`, and `jetpack-cloud-stage` environments already. It only makes sense to also enable this in the other environments that we use for internal and beta testing, `wpcalypso` and `horizon`.

ockham committed:
Revert "Gutenboarding: Add feature flag to redirect to Site Editor upon Site Creation (#43246)" (#43357)

This reverts commit afb8019a1e260fbb737fb76ee5fde57a17f5c78d.

bluefuton committed:
Comments: style Gutenberg captions (#43274)

p-jackson committed:
Nav Sidebar: show 10 most recently edited posts/pages in sidebar (#42606)

renovate[bot] committed:
chore(deps): update yarn to ^1.22.4 (#43314)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
fix(deps): update dependency cookie-parser to ^1.4.5 (#43358)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
chore(deps): update node.js (#43366)

* chore(deps): update node.js
* Update circleci image

Co-authored-by: Renovate Bot <bot@renovateapp.com>
Co-authored-by: Jon Surrell <jon.surrell@automattic.com>

alshakero committed:
Plans grid design tweaks (#43283)

lamosty committed:
P2: use .p2.blog subdomain for new sites (#43333)

lamosty committed:
Revert "P2: use .p2.blog subdomain for new sites (#43333)" (#43371)

This reverts commit 92537e5bde4d2d89891ed38cf0f23daba2882a2e.

Commit:
Gutenboarding: update plans and domains behaviour (#43321)

Co-authored-by: Omar Alshaker <omar@omaralshaker.com>

sgomes committed:
Extract and modularise `ui/reader` to `reader-ui` (#43343)

This removes Reader state from `ui`, while avoiding all of Reader being
loaded on boot.

Also extracts a few inline selectors into a proper selector module.

cpapazoglou committed:
Search: focus back on input when clicking close (#43364)

Co-authored-by: cpap <papazoglou.charalampos@gmail.com>

yansern committed:
Gutenboarding: Add hesitation nudge on intent gathering. (#43317)

* Add hesitation nudge on intent gathering.

* Added 3s delay before showing hint, 1s delay before hiding hint.

* Translation explantion.

* Rework using transition to fix issue where input hint suddenly appears on first keystroke when fade-in animation hasn't completely kicked in yet.

ockham committed:
Gutenboarding: Add feature flag to redirect to Site Editor upon Site Creation (take two) (#43246)

This PR watches out for a feature flag, and if it is set:
- Upon site creation, it sets the `site_creation_flow` option to `gutenboarding-site-editor` (rather than just `gutenboarding`)
- It takes the user to the site editor rather than the post (content) editor.

This reverts commit cb5986f2d874aef384b3ab9120fc974f50acee37.

mattgawarecki committed:
Jetpack Cloud: Fill the Plan sidebar icon for paids plan or products (#43300)

* Jetpack Cloud: Fill in the Plan sidebar icon for any paid plan or purchase of a Jetpack product

* Render QuerySitePurchases to account for sites with individual product purchases

klimeryk committed:
Revert "Domains: Update multi level TLD list (#43351)" (#43384)

This reverts commit ab96ff5ce4aaef4f96be6b67d6648cca54b30333.

sgomes committed:
Fix wrong key being used for preferences state. (#43385)

mattgawarecki committed:
Jetpack Cloud: Add feature flags to toggle the Jetpack section by site type (#43350)

* Jetpack Cloud: Add feature flags to toggle the Jetpack section by site type

* Create isJetpackSectionEnabledForSite selector

* Create renderNotFound middleware in index.web.js

* Jetpack Cloud: Check jetpack/features-section/* flags for site before navigating

* Check jetpack/features-section/* flags before choosing redirect target

* Check jetpack/features-section/* flags before showing Jetpack sidebar section

* Check jetpack/features-section/* flags before showing new Jetpack Cloud sidebar

* Only show new Jetpack Security settings pane if jetpack/features-section/* flags are enabled

* Only redirect to new Jetpack Security settings pane if jetpack/features-section/* flags are enabled

* Only show Business upsell if Jetpack section is enabled for site

* Change variable name to 'shouldPromoteJetpack' since that's what it's doing

* Remove 'render' prefix from middleware, since it doesn't actually render

annemirasol committed:
e2e: Add invites cleanup after tests (#43327)

* Add invites cleanup after test run

* Temporarily increase wait time

* Restore original wait ms

* Ensure email addr unique between two editor tests

* Move viewer cleanup to after hook, use cleanup flags

cleacos committed:
Remove is-wide-layout class from the Backup/Scan upsell (#43386)

rcanepa committed:
Jetpack section: add Automated Transfer step for Business sites (#43291)

* Create WPCOM Business to Automated Transfer

* Create the 'What is Jetpack?' footer component

* Add Jetpack Backup and Jetpack Scan illustrations

* Make it work for both Backup and Scan

* Add the new controller to all Backup routes

* Enable the controller for Scan's routes

* Remove '.' for string reuse

* Check if site is eligible for AT

* Merge UpSell and BusinessAT controllers

* Query eligibility from the sidebar

* Make the CTA trigger the AT

* Minify the Jetpack Backup illustration

Commit:
Combine Activity Logs (#43306)

vindl committed:
Site Editor: disable demo templates on stickered sites (#43334)

We don't need demo templates to be active any longer since
they might interfere with templates that FSE themes provide.

klimeryk committed:
Choose the correct additional info TLD form for second level TLDs (#43391)

This fixes what #43351 was aiming to, as well as the bug that was uncovered because of it (and probably dating all the way back to #20750?), which was the reason for the revert in #43384

gwwar committed:
Earn: add feature flag for safe rename of Simple Payments and Recurring Payments blocks (#43360)

yansern committed:
Domain Picker: Nudge hint tweaking (#43390)

* Don't hide the hint nudge when there is a site title value.

Co-authored-by: Razvan Papadopol <razvan.papadopol@automattic.com>

ianstewart committed:
FSE Plugin: New (updated) patterns (#42859)

* FSE Plugin: New (updated) patterns

This PR should add 10 new patterns to the FSE Plugin. Replaces https://github.com/Automattic/wp-calypso/pull/42175

* FSE Plugin: temporarily removing this pattern

Temporarily removing this pattern until we determine why it's not rendering correctly on WP.com. Don't want it to be a blocker on other patterns getting deployed! See https://github.com/Automattic/wp-calypso/pull/42859#issuecomment-641036139

* Add new patterns to categories

* Add "List" category and rearrange category assignments

Co-authored-by: Philip Jackson <p-jackson@live.com>

sixhours committed:
Buttons: Fix vertical alignment (#43405)

Commit:
Remove jetpack/scan from WPCOM sites (#43397)

Commit:
Update/backup view for wordpress com (#43407)

bluefuton committed:
Reader Lists: add plumbing for adding feeds, sites and tags to lists (#43173)

* Add data layer and actions

* Remove error notice

* Back to errorNotice - let's make sure the endpoint errors are translated instead

* hook up add button

* drop the console error

Co-authored-by: Jason Moon <4044428+jsnmoon@users.noreply.github.com>

* Re-add form and list items after rebase

* handle receiving a new feed id from the response

* Fix the 'feed not found' error notice

Co-authored-by: Ben Lowery <ben.lowery@automattic.com>
Co-authored-by: Ben Lowery <blowery@users.noreply.github.com>
Co-authored-by: Jason Moon <4044428+jsnmoon@users.noreply.github.com>

ramonjd committed:
Gutenboarding: add new template designs (#42354)

* initial commit: testing the template settings

* Updating fonts based on what I'm seeing in the fonts network traffic

* Adding a slot for Twenty Twenty

* rebase and adding 2 x new templaters

* Added the remaining templates/themes/designs ?

* Adding preview template images

* When the "gutenboarding/edge-templates" feature flag is true, show the new, probably not-yet-working templates.
Turned on for dev, stage, wpcalypso and horizon environments. That is, NOT PROD :)

Removing explicit reference to `featured` property in designs object usage. We might one day want to use various collection in that object, but I did it mainly to satisfy the requirements of the feature flag :)

regenerating the design screenshots

Regenerating at 900w at 72% quality

Removed twenty twenty

* Update stage.json

* update rockfield preview

* Updated font name case

* Renaming is_edge to is_alpha to describe new Gutenboarding designs, and to make sure we don't confuse labels with Gutenberg edge

andrewserong committed:
Add separate style for focus state of skip buttons so that it's more visible when tabbing via keyboard (#43419)

alshakero committed:
Add placeholder to site title input (#43225)

scinos committed:
Deduplicate yargs-parser (#43370)

scinos committed:
Deduplicate schema-utils (#43372)

scinos committed:
Update node on CI (#43368)

* Update node on CI

* Fix typo in CI config

* Run correct command

* Install nvm without running a shell script

lsl committed:
Add error handling for new site creation failure modes (#43092)

* Add create-site-error component to handle errors

While unlikely, the create site endpoint may fail.

In this scenario users are pushed back to the previous onboarding step
without notification of any failure mode ocurring.

This PR adds an interim error step to provide feedback and give users a
way to break out of a potentially broken flow.

* Hide default header text during CreateSite transition

daledupreez committed:
Add social login information to security checkup (#43226)

Authored-by: Dale du Preez <dale@automattic.com>

renovate[bot] committed:
fix(deps): update dependency yargs to ^15.3.1 (#43374)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

scinos committed:
Deduplicate run-async (#43424)

cpapazoglou committed:
Post List Block: make sure button doesn't shrink beyond its content (#43426)

Co-authored-by: cpap <papazoglou.charalampos@gmail.com>

lamosty committed:
P2: use .p2.blog subdomain for new sites (#43379)

hambai committed:
Update the font size of the domain navigation item subtitle (#43316)

Update the font-size of the domain navigation item subtitle to be 14px ($font-body-small)

sgomes committed:
Modularise checklist state (#42712)

Fixes #42426

sgomes committed:
Modularise googleMyBusiness (#43325)

Also moves `ui/google-my-business` under the main state key
for `googleMyBusiness`.

Fixes #42436

sgomes committed:
Modularise memberships state (#42955)

* Modularise memberships state

Also refactors inline selectors, moving them to dedicated modules, and removes
some unused selectors.

Fixes #42457.

* Use consts for default values.

sgomes committed:
Modularise application passwords state (#43100)

Fixes #42422.

claudiucelfilip committed:
Signup test: Use Passwordless signup after Plans step (#43061)

* Initialize the AB test and flow

* Set passwordless singup form to be loaded on the tested flow

* Make Google and Apple login buttons support customizable labels.

This will be needed to make the passwordles Singup page match the intened look.

* Add custom prop overrides for the passwordless signup step

* Update the passwordless Signup layout and copy to match the desired variant look

This expands the customizability of the components so they received the needed texts from the step configuration.

* Hide Back to Sites button on domain step

There should still be a back button when choosing an already owned domain.

* Add missing React dependency and remove unnecessary code

* Add copy changes for the passwordless signup step.

* Simplify passwordless signup flow and step naming

* Remove unnecessary ab test check

* Remove unnecessary defaulting en localeTarget for the ab test

* Remove unnecessary custom footer text from singup form

* Remove unnecessary dependency

* Set custom copy for footer login link

* Update flow details and add it to domain test whitelist

* Update ab test timestamp

* Add marketing price group as provided dependency for passwordless signup

It defaults to empty string and is needed by the social signup.

* Prevent passwordless onboarding flow from kicking in on direct-to-plan flows

jsnajdr committed:
Remove isCalypsoClient variable from webpack.config.js (#43378)

* Remove isCalypsoClient variable from webpack.config.js

* Register the Lodash replacement plugin as the last one, to ensure earlier module replacements are processed

sgomes committed:
Clarify some points around modularized state (#43439)

- Distinguish between legacy Redux reducers and modularized ones
- Update preferred folder structure
- Explain that modularization happens at the top level
- Clarify which UI state should live under `ui/`
- Minor improvements and fixes

sixhours committed:
Typography: Use npm package import for Recoleta (#43338)

sirreal committed:
E2E: Fix Gutenberg editor test (#43440)

* Trigger a second click if necessary to open block inserter

lamosty committed:
P2: enable the P2 signup flow on every platform (#43445)

Commit:
Jetpack: fix and standardize headings styles in backup and scan (#43409)

sirbrillig committed:
Checkout: Make sure PayPalButtonContents always returns a component (#43402)

This may not be necessary, but we have seen strange errors where React
seems to be getting confused about the contents of the DOM when removing
nodes. It's possible this is because PayPalButtonContents sometimes
returns a string and sometimes returns a component.

mattwondra committed:
Add Tracks events for HappOps (#43416)

* Add Tracks event when a user closes their account

* Add Sibyl trackers for first click and whether suggestions are showing on submit

sarayourfriend committed:
State: Add transient media items reducer (#42999)

* State: Add transient media items reducer

* State: Account for transient items in when selecting a media item

* Add accidentally ommited line in tests

* Remove potentially confusing ?? operator usage

Commit:
Jetpack Cloud: Switch to New Menu (#43406)

mattwondra committed:
Remove Limited Availability notice for Chat (#43447)

alshakero committed:
Refactor and reorder domain suggestions in domain picker (#43373)

rcanepa committed:
Jetpack section: add switch to control when we display the Business AT (#43413)

* Control when we show the Business AT component

* Pass down the Automated Transfer Status

* Fix infinite loop on blocked AT sites

* Improve types

* Don't display the BusinessAT if the AT has been completed

noahtallen committed:
Run phpcbf on fse plugin (#43452)

victorespigares committed:
AB Test for Business bump offer (#43199)

* AB test and updated copy.

* Do not show upsell offer nudge if another offer was previously shown

* Do not show upsell offer nudge if another offer was previously shown

* remove empty cart feature, not needed for now

* In the new checkout flow, don't assign to the test for failed transactions

* Re-added the test with hideNudge prop

* Adding Business bump offer

* Removing unused variable

* Added Concierge offer for all Premium customers in control group

* Removing unused variable

* Adding control group gets Concierge offer for Composite Checkout

* Removing merged leftover

* Fixing lint errors

Co-authored-by: Niranjan Uma Shankar <niranjan.uma.shankar@automattic.com>

eltongo committed:
Fix missing key attribute warning for Manage All Domains link (#43456)

Commit:
Promo card CTA: simplify invalid prop removal

Commit:
Merge branch 'fix/jetpack-self-target-warning' of github.com:Automattic/wp-calypso into fix/jetpack-self-target-warning

monsieur-z commented:
Some Git mishap here. Let me open a new PR instead. I'll link this one.

a8ci18n commented on apps/full-site-editing/full-site-editing-plugin/wpcom-block-editor-nav-sidebar/src/components/nav-sidebar/index.tsx:
```diff
@@ -130,23 +137,81 @@ export default function WpcomBlockEditorNavSidebar() {
+			<Button className={ buttonClasses }>
+				<div className="wpcom-block-editor-nav-sidebar__title-container">
+					<div className={ titleClasses }>
+						{ item.title?.raw || __( 'Untitled', 'full-site-editing' ) }
```
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **40** times:
`translate( 'Untitled', { context: 'post title for posts with no title'} )` *ES Score: 16*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/fix/jetpack-self-target-warning)

monsieur-z commented:
Follow up in #43483 43483
